### PR TITLE
Handle Docker NotFound exception.

### PR DESCRIPTION
**Pull request by acaprez**
When DOCKER_HOST is set and points to a tcp address,
docker.APIClient.pull() returns a docker.errors.NotFound
exception instead of a JSON object. Catch this gracefully and
raise a CacheMissException() to be consistent with the JSON case
when the tag isn't found.

xref #95

### CODE REVIEW

**Comment by graingert** [comment]
can you squash your commits, and add the `Fixes ` directive to the commit.

File: shipwright/cache.py [hunk 1]
```diff
@@ -3,6 +3,7 @@
 import sys
 import traceback
 
+from docker import errors as d_errors
 from requests import exceptions as requests_exceptions
```
Comment (graingert) [inline shipwright/cache.py]:
```python
def pull(client, *args, **kwargs):
    try:
        for evt in client.pull(*args, **kwargs):
            yield compat.json_loads(evt)
    except d_errors.NotFound as e:
        yield {'error': e.explanation, 'errorDetail': {'message': e.explanation}}
```

Comment (graingert) [inline shipwright/cache.py]: then we can just change `client.pull(` to `pull(client, `

Comment (acaprez) [inline shipwright/cache.py]: Yep, that would be cleaner.  Should that be incorporated here, or a separate PR?

Comment (graingert) [inline shipwright/cache.py]: Yeah in this PR. Just add it to this file for now

**Pull request merged**